### PR TITLE
[MAINTENANCE] Update testing framework to work with bigquery

### PR DESCRIPTION
**Pull request by tyler-hoffman**
These are the first bigquery tests (other than docs tests), so there are a couple config type things I had to add. Sorry!

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: constraints-dev.txt [hunk 1]
```diff
@@ -5,27 +5,3 @@
...
-#		TypeError: Descriptors cannot not be created directly.
-#		If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
-#		If you cannot immediately regenerate your protos, some other possible workarounds are:
-#		 1. Downgrade the protobuf package to 3.20.x or lower.
-#		 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
-#
-#		More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
-# google-cloud-bigquery has a fix for a refactor to shapely in their 3.3.6 release
-#       https://github.com/googleapis/python-bigquery/pull/1376
-# With the new requirement of google-cloud-bigquery>=3.3.6, there is a hard conflict with protobuf==3.20
```
Comment (tyler-hoffman) [inline constraints-dev.txt:20]: This was causing pip to run forever and eventually error out in python 3.12. See https://github.com/great-expectations/great_expectations/actions/runs/11863600908/job/33067926751. Note that it was fine on 3.9.

But either way, this constraint was added about 2 years ago. google-cloud-bigquery is now on 3.27, and protobuf is on 5.28. I think we're safe to remove this.

Comment (cdkini) [reply constraints-dev.txt:20]: Yeah I think we're good to remove this - does this cause any semver updates? I would think not since it's an optional dep?

Comment (tyler-hoffman) [reply constraints-dev.txt:20]: I don't believe this needs semver updates.

File: tests/integration/test_utils/data_source_config/big_query.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    @cached_property
+    def big_query_connection_config(self) -> BigQueryConnectionConfig:
+        return BigQueryConnectionConfig()  # type: ignore[call-arg]  # retrieves env vars
+
+
+class BigQueryConnectionConfig(BaseSettings):
+    GE_TEST_GCP_PROJECT: str
+    GE_TEST_BIGQUERY_DATASET: str
+    GOOGLE_APPLICATION_CREDENTIALS: str
```
Comment (cdkini) [inline tests/integration/test_utils/data_source_config/big_query.py:86]: These are set in CI? If so, are they done as secrets or vars? Could we document that here?

File: tests/conftest.py [hunk 3]
```diff
@@ -115,6 +115,7 @@
     "aws_creds",
     "aws_deps",
     "big",
+    "bigquery",
```
Comment (cdkini) [inline tests/conftest.py:118]: What impact does this have? We already have this marker but what changes by adding to this list?

**Review by cdkini — APPROVED** [review]

File: tests/conftest.py (hunk 3 above)
Comment (tyler-hoffman) [reply tests/conftest.py:118]: We have checks that fail without it: https://github.com/great-expectations/great_expectations/actions/runs/11862352884/job/33061555346

File: tests/integration/test_utils/data_source_config/big_query.py (hunk 2 above)
Comment (tyler-hoffman) [reply tests/integration/test_utils/data_source_config/big_query.py:86]: Updated, thanks!

**Pull request merged**